packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -86,49 +89,186 @@ impl<C: AsClient> PostgresStore<C> {
-                //             &link,
-                //             dependency_context.change_depth(GraphResolveDepths {
-                //                 link_resolve_depth: dependency_context
-                //                     .graph_resolve_depths
-                //                     .link_resolve_depth
-                //                     - 1,
-                //                 ..dependency_context.graph_resolve_depths
-                //             }),
-                //         )
-                //         .await?;
-                //     }
-                // }
+                for outgoing_link_entity in <Self as Read<Entity>>::read(
+                    self,
+                    &Filter::for_outgoing_link_by_source_edition_id(entity_edition_id),
+                )
+                .await?
+                {
+                    // We want to log the time the link entity was *first* added from this entity.
+                    // We therefore need to find the timestamp of the first link entity
+                    // TODO: this is very slow, we should update structural querying to be able to
+                    //  get the first timestamp of something efficiently
+                    let mut all_outgoing_link_entity_editions: Vec<_> =
+                        <Self as Read<Entity>>::read(
+                            self,
+                            &Filter::for_entity_by_entity_id(
+                                outgoing_link_entity.metadata().edition_id().base_id(),
+                            ),
+                        )
+                        .await?
+                        .into_iter()
+                        .map(|entity| entity.metadata().edition_id())
+                        .collect();
+
+                    all_outgoing_link_entity_editions.sort();
+
+                    let earliest_version = all_outgoing_link_entity_editions
+                        .into_iter()
+                        .next()
+                        .expect(
+                            "we got the edition id from the entity in the first place, there must \
+                             be at least one version",
+                        )
+                        .version();
+
+                    self.resolve_entity_dependency_with_edge_kind(
+                        &mut dependency_context,
+                        entity_edition_id,
+                        outgoing_link_entity.metadata().edition_id(),
+                        GenericOutwardEdge {
+                            // (HasLeftEndpoint, reversed=true) is equivalent to an
+                            // outgoing `Link` `Entity`
+                            kind: KnowledgeGraphEdgeKind::HasLeftEndpoint,
+                            reversed: true,
+                            endpoint: EntityIdAndTimestamp::new(
+                                outgoing_link_entity.metadata().edition_id().base_id(),
+                                earliest_version.inner(),
```

Comment:
We've discussed this together as well and went through the (valid) reasoning for why the vertex timestamp is the "created at" timestamp, but it does not seem like the timestamps in the vertex set and the edge set should be "incompatible". 

When querying the subgraph at the latest version, we could end up with a vertex that has a timestamp which is not present in the vertex. This isn't a problem, but it is slightly unintuitive. Perhaps having the "created at" on the link would be more consistent, but I'm not sure what the implications for the rest of the subgraph is if we do this.